spec/examples/metric_example.rb

```diff
@@ -31,7 +31,7 @@
 
   describe '.new' do
     it 'returns a new metric' do
-      expect(subject).to be
+      expect(subject).to be_a(Prometheus::Client::Metric)
```

Comment:
Since we're now leaning into the idea that `Prometheus::Client::Metric` provides a bunch of base metric behaviour which is tested independently, make sure that other metrics actually subclass it.